src/rdpq/rdpq_tex.c

```diff
@@ -293,6 +298,34 @@ static void texload_tile(tex_loader_t *tload, int s0, int t0, int s1, int t1)
     rdpq_set_tile_size_fx(tload->tile, s0, t0, s1, t1);
 }
 
+
+static void texload_settile(tex_loader_t *tload, int s0, int t0, int s1, int t1)
+{
+    tex_format_t fmt = surface_get_format(tload->tex);
+
+    rdpq_set_tile(tload->tile, fmt, tload->tmem_addr, tload->rect.tmem_pitch, &(tload->tileparms));
+
+    s0 = s0*4 + tload->rect.s0fx;
+    t0 = t0*4 + tload->rect.t0fx;
+    s1 = s1*4 + tload->rect.s1fx;
+    t1 = t1*4 + tload->rect.t1fx;
+    rdpq_set_tile_size_fx(tload->tile, s0, t0, s1, t1);
+}
+
+static void texload_settile_offset(tex_loader_t *tload, int s0, int t0, int s1, int t1, int tmem_offset)
+{
+    assertf(tmem_offset % 8 == 0, "Offset %i must be in multiples of 8", tmem_offset);
+    tex_format_t fmt = surface_get_format(tload->tex);
+
+    rdpq_set_tile(tload->tile, fmt, tmem_offset, tload->rect.tmem_pitch, &(tload->tileparms));
```

Comment:
shouldn't this be `tload->tmem_addr + tmem_offset`? Also how does this work if `tmem_addr` is already precomputed as part of the autotmem allocator?